sunpy/visualization/imageanimator.py

```diff
@@ -878,3 +875,112 @@ def update_plot(self, val, line, slider):
         if val != slider.cval:
             line.set_ydata(self.data[self.frame_slice])
             slider.cval = val
+
+class ImageAnimatorWCS(ImageAnimator):
+    """
+    Create a matplotlib backend independent data explorer for 2D images.
+
+    The following keyboard shortcuts are defined in the viewer:
+
+    - 'left': previous step on active slider
+    - 'right': next step on active slider
+    - 'top': change the active slider up one
+    - 'bottom': change the active slider down one
+    - 'p': play/pause active slider
+
+    This viewer can have user defined buttons added by specifying the labels
+    and functions called when those buttons are clicked as keyword arguments.
+
+    Parameters
+    ----------
+    data: ndarray
+        The data to be visualized >= 2D
+    
+    wcs: sunpycube.wcs_util.WCS
+        The wcs data.
+
+    image_axes: list
+        The two axes that make the image
+
+    fig: mpl.figure
+        Figure to use
+
+    axis_ranges: list of physical coordinates for array or None
+        If None array indices will be used for all axes.
+        If a list it should contain one element for each axis of the numpy array.
+        For the image axes a [min, max] pair should be specified which will be
+        passed to :func:`matplotlib.pyplot.imshow` as extent.
+        For the slider axes a [min, max] pair can be specified or an array the
+        same length as the axis which will provide all values for that slider.
+        If None is specified for an axis then the array indices will be used
+        for that axis.
+
+    interval: int
+        Animation interval in ms
+
+    colorbar: bool
+        Plot colorbar
+
+    button_labels: list
+        List of strings to label buttons
+
+    button_func: list
+        List of functions to map to the buttons
+
+    unit_x_axis: astropy.units
+        The unit of x axis.
+
+    unit_y_axis: astropy.units
+        The unit of y axis.
+
+    slices_wcsaxes: tuple
+        Slices parameter of WCSAxes. Example: (0, 'y', 'x') for 3D Data
+    Extra keywords are passed to imshow.
+
+    """
+    def __init__(self, data, wcs=None, image_axes=[-2, -1], unit_x_axis=None, unit_y_axis=None, slices_wcsaxes=None, axis_ranges=None, **kwargs):
```

Comment:
Right.  So the first image_axes value is the x axis and the second is the y axis?  Then you just find the corresponding axes in the wcs object.